poetry/console/commands/init.py

```diff
@@ -66,9 +67,26 @@ def handle(self):
         from poetry.utils.env import SystemEnv
         from poetry.core.vcs.git import GitConfig
 
+        original_toml = None
+
         if (Path.cwd() / "pyproject.toml").exists():
-            self.line("<error>A pyproject.toml file already exists.</error>")
-            return 1
+            with (Path.cwd() / "pyproject.toml").open() as toml_file:
+                original_toml = tomlkit.loads(toml_file.read())
+
+                try:
+                    if original_toml["tool"]["poetry"]:
+                        self.line(
+                            "<error>A pyproject.toml file with a poetry section already exists.</error>"
+                        )
+                        return 1
+                except KeyError:
+                    pass
+
+                if original_toml.get("build-system"):
+                    self.line(
+                        "<error>A pyproject.toml file with a defined build-system already exists.</error>"
+                    )
+                    return 1
```

Comment:
```suggestion
                if original_toml.get("tool", {}).get("poetry) or original_toml.get("build-system"):
                    self.line(
                        "<error>A pyproject.toml file with a poetry section already exists.</error>"
                    )
                    return 1
```

Should we also handle cases where `tool.poetry` exists and `build-system` does not?

Reply:
Nice idea to set the default return value for `get` to an empty dict!

I would prefer to keep the checks for  `tool.poetry` and `build-system` separated to give a proper error message.

> Should we also handle cases where tool.poetry exists and build-system does not?

Shouldn't the code above already handle this? Whenever one of these both is in the `pyproject.toml` it should quit.